static/app/views/alerts/rules/issue/details/ruleDetails.tsx

```diff
@@ -61,9 +63,47 @@ class AlertRuleDetails extends AsyncComponent<Props, State> {
     });
   }
 
-  componentDidUpdate(prevProps: Props) {
+  async componentDidUpdate(prevProps: Props) {
     const {organization: prevOrg, params: prevParams} = prevProps;
-    const {organization: currOrg, params: currParams} = this.props;
+    const {organization: currOrg, params: currParams, location} = this.props;
+
+    const {ruleId, projectId: projectSlug} = currParams;
+
+    const rule = this.state.rule;
+    const urlParams = new URLSearchParams(location.search);
+    if (urlParams.get('mute') === '1' && rule && !rule.snooze) {
+      try {
+        await this.api.requestPromise(
```

Comment:
This looks like it uses a lot of the same logic that's inside the SnoozeAlert component. I think it might be better to move it there so the logic is colocated. It also means that it will be properly gated by the feature flag as well.

You should be able to do this with `useLocation()` and a useEffect.

Reply:
@malwilley its a class component

Reply:
oh I meant you could use these hooks if this logic was in `snoozeAlert.tsx` which is an FC